curp/src/server.rs

```diff
@@ -764,6 +847,55 @@ impl<C: 'static + Command, CE: 'static + CommandExecutor<C>> Protocol<C, CE> {
             state.term,
         )))
     }
+
+    /// Handle `Vote` requests
+    #[allow(clippy::pedantic)] // need not return result, but to keep it consistent with rpc handler functions, we keep it this way
+    fn vote(
+        &self,
+        request: tonic::Request<VoteRequest>,
+    ) -> Result<tonic::Response<VoteResponse>, tonic::Status> {
+        let req = request.into_inner();
+        debug!(
+            "vote received: term({}), last_log_index({}), last_log_term({}), id({})",
+            req.term, req.last_log_index, req.last_log_term, req.candidate_id
+        );
+
+        // just grab a write lock because it's highly likely that term is updated and a vote is granted
+        let mut state = self.state.write();
+
+        // calibrate term
+        match req.term.cmp(&state.term) {
+            Ordering::Less => {
+                return Ok(tonic::Response::new(VoteResponse::new_reject(state.term)));
+            }
+            Ordering::Equal => {}
```

Comment:
If we get a vote request with the same term from another leader while we're still a leader, what will happen?

Reply:
Should there be only one leader per term?

Reply:
You're handling a vote here, and maybe that peer is trying to be a leader because it doesn't know the current one.

Reply:
Then it will reject if it is a leader, because when its `voteFor` must be itself(set when it starts the election).